components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/api/APIMgtAdminService.java

```diff
@@ -122,4 +122,21 @@
      * @throws APIManagementException If failed to register labels.
      */
     void registerGatewayLabels(List<Label> labels, String overwriteLabels) throws APIManagementException;
+
+    /**
+     * Retrieve API's gateway configuration
+     *
+     * @param apiId     UUID of the API
+     * @return          The API gateway configuration
+     * @throws APIManagementException   If failed to retrive API gateway config
+     */
+    String getAPIGatewayServiceConfig(String apiId) throws APIManagementException;
+
+    /**
+     *
```

Comment:
Shall we please use a more specific exception which extends from APIManagementException?